src/Native/Runtime/arm/PInvoke.S

```diff
@@ -30,46 +18,53 @@
 // Also, the codegenerator must ensure that there are no live GC references in callee saved registers.
 //
 NESTED_ENTRY RhpPInvoke, _TEXT, NoHandler
-	        PROLOG_PUSH "{r4-r7,lr}"
+        str         lr, [r0, #OFFSETOF__PInvokeTransitionFrame__m_RIP]
+        str         r7, [r0, #OFFSETOF__PInvokeTransitionFrame__m_FramePointer]
+        str         sp, [r0, #OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs]
+
+        PROLOG_PUSH "{r5-r7,lr}"
 
-	        str         lr, [r0, #OFFSETOF__PInvokeTransitionFrame__m_RIP]
-	        str         r7, [r0, #OFFSETOF__PInvokeTransitionFrame__m_FramePointer]
-	        str         sp, [r0, #OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs]
+        // get TLS global variable address
+        mov         r5, r0
+        ldr         r0, .TLS_CURRENT_THREAD_LABEL
+.TLS_CURRENT_THREAD_CODE:
+        add         r0, pc, r0
+        bl          __tls_get_addr(PLT)
 
-	        mov	        r5, r0
-	        // R0 = GetThread(), R6 - Trash Register
-	        INLINE_GETTHREAD r6
-	        str         r0, [r5, #OFFSETOF__PInvokeTransitionFrame__m_pThread]
-	        str         r5, [r0, #OFFSETOF__Thread__m_pTransitionFrame]
+        str         r0, [r5, #OFFSETOF__PInvokeTransitionFrame__m_pThread]
+        str         r5, [r0, #OFFSETOF__Thread__m_pTransitionFrame]
 
```

Comment:
There is mix of tabs and spaces. Could you please clean up the formatting? (We prefer spaces.)